flink-runtime/src/main/java/org/apache/flink/runtime/management/JMXService.java

```diff
@@ -85,6 +86,9 @@ private static JMXServer startJMXServerWithPortRanges(Iterator<Integer> ports) {
 		while (ports.hasNext() && successfullyStartedServer == null) {
 			JMXServer server = new JMXServer();
 			int port = ports.next();
+			if (port == 0) { // try poke with a random port when port is set to zero
+				port = tryPokeForNewPort();
```

Comment:
use `NetUtils#getAvailablePort` instead; note that this approach is inherently unreliable.

Reply:
Good point. This approach is yes in fact unreliable. and consider # of containers our typical prod use case has, it's very likely that >1 of them will hit the corner case.